### PR TITLE
Add projects section into devfile spec document

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Adds new `Adding projects to a devfile` section into devfile specification chapter.
Adds docs for `sparseCheckoutDir` feature.
Fixes some indents and `clonePath` parameters position.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13671

### CODE REVIEW

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc [hunk 1]
```diff
@@ -3,6 +3,94 @@
...
+Project source consists of two mandatory values: `type` and `location`.
+
+`type` defined kind of project source provider.
+ +
+`location` contains URI to where project source is located.
+
+Che support the following project types:
+
+ * `git` - projects which is under git VCS. Location should point to clone link.
+ * `guthub` - the same as above, but for projects hosted on https://github.com/[GitHub] only.
```
Comment (AndrienkoAleksandr) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]: guthub -> github

Comment (AndrienkoAleksandr) [inline src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]: I guess You wanted to create markdown link:
https://github.com/[GitHub] -> [https://github.com/](GitHub)

**Comment by AndrienkoAleksandr** [comment]
Great job!

**Review by AndrienkoAleksandr — APPROVED** [review]

File: src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc (hunk 1 above)
Comment (mmorhun) [reply src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]: Will fix, thanks

Comment (mmorhun) [reply src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc]: I used ascii doc format

**Comment by mmorhun** [comment]
@rkratky please take a look

**Comment by mmorhun** [comment]
I would like to merge implementation with documentation. @rkratky please take a look at this one...

**Review by rkratky — CHANGES_REQUESTED** [review]
@mmorhun, apologies for the delay. I submitted a PR (#860) against your branch with suggested language and formatting improvements.

**Review by rkratky — APPROVED** [review]

**Comment by rkratky** [comment]
@mmorhun, please merge whenever you're ready.

**Pull request merged**